### DISCUSSION TITLE
Split file in client-read-format-h264-save-to-disk example

### DISCUSSION THREAD

**Discussion by roberto497** [bluenviron/gortsplib#321] [Questions]
It is possible to create a new file every x seconds of captured video?

**Comment by aler9** [comment]
Hello, there's an example on how to generate splitted, periodic MPEG-TS files inside the HLS muxer:
https://github.com/bluenviron/gohlslib/blob/main/muxer_segmenter_mpegts.go

Basically you can measure the period between frames and split the file whenever the period is greater than a threshold and when the current frame is an I-Frame.

Keep in mind that the primary aim of gortsplib is to provide a RTSP library, not to provide a complete media library, therefore examples are meant for displaying simple integrations, and file-related features are outside the scope of this library.
